compiler/modules/front.syntax/parseExpr.bal

```diff
@@ -224,14 +224,23 @@ function startPrimaryExpr(Tokenizer tok) returns Expr|err:Syntax {
     if t is [IDENTIFIER, string] {
         Position pos = tok.currentPos();
         string identifier = t[1];
-        VarRefExpr expr = { varName: identifier };
         check tok.advance();
-        t = tok.current();
-        if t == "(" {
+        string varName;
+        string? prefix;
+        if tok.current() == ":" {
```

Comment:
Something similar to this bit of code is repeated in several places. Try factoring out a function:

`function parseOptQualIdentifier(Tokenizer tok, string identifier) returns [string?, string]|err:Syntax;`

Then this code would become

`var [prefix, varName] = check parseOptQualIdentifier(tok, t[1]);`